tests/unit/symbolic_regression/agraph/test_agraph_mutation.py

```diff
@@ -315,6 +330,7 @@ def test_impossible_param_or_prune_mutation(mocker, algo_index,
     c_stack = child.mutable_command_array
 
     np.testing.assert_array_equal(c_stack, p_stack)
+    assert mutation.last_mutation_type == mutation.types[algo_index]  # TODO should be None?
```

Comment:
These test cases test when no mutation happens. Before the EA changes, we would set the last_mutation_type even if no mutation occurred (e.g., prune mutation got called but couldn't find a suitable subtree to prune), is this still the expected behavior?